src/app/work-item/work-item.service.ts

```diff
@@ -687,6 +692,18 @@ export class WorkItemService {
   }
 
   /**
+   * Usage: This method emit a new work item created event
+   * The list view and board view listens to this event
+   * and updates the view based on applied filters.
+   */
+
+  emitAddWI(workItem: WorkItem) {
+    if(this.filterService.doesMatchCurrentFilter(workItem)){
```

Comment:
This is okay, but think, now work item service has a dependency on filter service. We can have it in the component. No matter what item it is, it should be emitted and then from the component we check if the item matches current filter or not.
Also, we can use the same observer in other places where filter doesn't matter.